dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -1005,4 +1009,146 @@ public void resetSecrets(final User user)
        appsCache.clearCache();
     }
 
+    /**
+     *
+     * @param key
+     * @param paramAppKeysBySite
+     * @return
+     */
+    public File exportSecrets(final Key key, final boolean exportAll,
+            final Map<String, Set<String>> paramAppKeysBySite, final User user)
+            throws DotDataException, DotSecurityException, IOException {
+
+        if(!userAPI.isCMSAdmin(user)){
+            throw new DotSecurityException("Only Admins are allowed to perform an export operation.");
+        }
+
+        final AppsSecretsImportExport exportedSecrets;
+        if (exportAll) {
+            exportedSecrets = collectSecretsForExport(appKeysByHost(), user);
+        } else {
+            exportedSecrets = collectSecretsForExport(paramAppKeysBySite, user);
+        }
+
+        final File tempFile = File.createTempFile("secretsExport", ".tmp");
+        try {
+            writeObject(exportedSecrets, tempFile.toPath());
+            final byte[] bytes = Files.readAllBytes(tempFile.toPath());
+            try {
+                final File file = File.createTempFile("secrets", ".export");
+                final byte[] encrypted = AppsUtil.encrypt(key, bytes);
+                try (OutputStream outputStream = Files.newOutputStream(file.toPath())) {
+                    outputStream.write(encrypted);
+                    return file;
+                }
+            } catch (EncryptorException e) {
+                throw new DotDataException(e);
+            }
+        } finally {
+            tempFile.delete();
```

Comment:
just in case check NPE

Reply:
when I did so.. It says the condition is always different from null